File: src/main/java/me/cozease/bookstorebackend/service/BookService.java
package me.cozease.bookstorebackend.service;

import me.cozease.bookstorebackend.entity.Book;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

/**
 * 图书服务类
 */
@Transactional
public interface BookService {
    /**
     * 获取所有图书
     */
    List<Book> getAllBooks();

    /**
     * 根据ID获取图书
     */
    Optional<Book> getBookById(Long id);

    /**
     * 搜索图书（按标题、作者或出版社）
     */
    List<Book> searchBooks(String keyword);

    /**
     * 添加图书
     */
    Book addBook(Book book);

    /**
     * 更新图书
     */
    Book updateBook(Long id, Book bookDetails);

    /**
     * 删除图书
     */
    void deleteBook(Long id);

    /**
     * 检查图书库存
     */
    boolean checkStock(Long bookId, Integer quantity);

    /**
     * 更新图书库存
     */
    void updateStock(Long bookId, Integer quantity);
}

File: src/services/bookService.js
import request from './request';

export const BookService = {
    // 获取所有图书
    getAllBooks: async () => {
        return request('/books');
    },

    // 获取图书详情
    getBookById: async (id) => {
        return request(`/books/${id}`);
    },

    // 搜索图书
    searchBooks: async (keyword) => {
        return request(`/books/search?keyword=${encodeURIComponent(keyword)}`);
    },

    // 更新图书信息（管理员功能）
    updateBook: async (id, bookData) => {
        return request(`/books/${id}`, {
            method: 'PUT',
            body: JSON.stringify(bookData),
        });
    },

    // 添加新图书（管理员功能）
    addBook: async (bookData) => {
        return request('/books', {
            method: 'POST',
            body: JSON.stringify(bookData),
        });
    },

    // 删除图书（管理员功能）
    deleteBook: async (id) => {
        return request(`/books/${id}`, {
            method: 'DELETE',
        });
    },
};
